src/styles/calendar.scss

```diff
@@ -0,0 +1,59 @@
+.calendar-container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  max-width: $calendar-width;
+
+  &__day {
+    width: $square-side;
+    height: $square-side;
+    background-color: #eee;
+    box-sizing: border-box;
+    border: 1px solid #000;
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+    text-align: center;
+    justify-content: center;
+    padding-top: 35px;
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:first-child {
+      margin-left: calc(#{$calendar-width} - #{$square-side});
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $i from 1 through 31 {
+    .calendar__day:nth-child(#{$i})::before {
+      content: "#{$i}";
+    }
+  }
+  
+  @each $start-day, $switch in $week-days {
+    &--start-day--#{$start-day} {
+      .calendar__day:first-child {
+        margin-left: calc(#{$calendar-width} - #{$square-side} * (6 - $switch) - #{$gap} * $switch);
```

Comment:
why do you need to do (6 - $switch) where $switch is 0, and also you multiply $gap on 0, why? Generally it's so complicated, could you think how  make it easier